compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -671,6 +672,17 @@ fn encode_ty<'tcx>(
             typeid.push_str(&s);
         }
 
+        ty::CoroutineWitness(def_id, args, ..) => {
```

Comment:
You shouldn't need to encode these.

Reply:
Or transform them.

Reply:
Update: You can just encode the [`parent_args()`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/sty/struct.CoroutineClosureArgs.html#method.parent_args) of the containing coroutine closure to continue avoiding the need to encode a witness here.

Similarly for `Coroutine` you only need to encode `parent_args()` + the [`kind_ty()`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/sty/struct.CoroutineArgs.html#method.kind_ty).